src/Autofac/Core/IComponentRegistration.cs

```diff
@@ -83,10 +83,16 @@ public interface IComponentRegistration : IDisposable
         IResolvePipeline ResolvePipeline { get; }
 
         /// <summary>
-        /// Gets a value indicating whether the registration is a 1:1 adapter on top
-        /// of another component (e.g., Meta, Func, or Owned).
+        /// Gets the options for the registration.
         /// </summary>
-        bool IsAdapterForIndividualComponent { get; }
+        RegistrationOptions Options { get; }
+
+        /// <summary>
+        /// Tests whether a given option (or combined set of options) is enabled for the registration.
+        /// </summary>
+        /// <param name="option">The option.</param>
+        /// <returns>True if the option (or options) is enabled for the registration.</returns>
+        bool HasOption(RegistrationOptions option);
```

Comment:
I added the `HasOption` to give us a typed version, but you're right, a typed extension method might be a better choice anyway.

The boxing overhead is only true for .NET Framework; as I noted in an earlier comment, HasFlag has been fixed in .NET Core with a JIT Intrinsic:

**.NET 461**
|      Method |      Mean |     Error |    StdDev | Ratio | RatioSD |
|------------ |----------:|----------:|----------:|------:|--------:|
|    BoolProp |  2.856 ns | 0.0235 ns | 0.0220 ns |  1.00 |    0.00 |
| EnumHasFlag | 41.708 ns | 0.2398 ns | 0.2126 ns | 14.60 |    0.14 |
|  EnumBitOps |  2.982 ns | 0.0358 ns | 0.0318 ns |  1.04 |    0.01 |

**.NET Core 3.1**
|      Method |     Mean |     Error |    StdDev | Ratio | RatioSD |
|------------ |---------:|----------:|----------:|------:|--------:|
|    BoolProp | 3.382 ns | 0.0461 ns | 0.0409 ns |  1.00 |    0.00 |
| EnumHasFlag | 3.828 ns | 0.0398 ns | 0.0353 ns |  1.13 |    0.02 |
|  EnumBitOps | 3.959 ns | 0.0474 ns | 0.0396 ns |  1.17 |    0.01 |

That said, I suppose it's not fair to be mean to our netfx consumers to get that 0.1ns saving...I'll just switch to bit testing it.

Reply:
Weird; when overloading HasFlag via an extension method, the compiler picks the runtime's HasFlag. I thought extension methods got picked on the most specific type match. Might have to call it `HasOption`.

Reply:
Oh right right right I remember that now. My brain is fried.

I think not having it on the interface is valuable anyway; folks can check the already-exposed options object directly and if it's really that hard we can add an extension method for the interface since it'll be implemented identically in any implementation of this.

And, no, I still can't quite grasp "default implementations of methods on interfaces." I feel like that's... yeah, if folks want that, it's an abstract class. I'm not sure that was a feature I'd have voted for.